tools/python_tools/cuvslam_tools/dataset_preparation/rgbd.py

```
@@ -0,0 +1,484 @@
+    # Denominator converting raw depth samples to metres. Native TUM 16-bit PNG
+    # depth uses 5000; float32 depth already in metres uses 1.
+    depth_scale_factor: float
+
+
+def _parse_timestamp_ns(text: str, source: str, line_number: int) -> int:
+    try:
+        return int(Decimal(text) * NANOSECONDS_PER_SECOND)
+    except (InvalidOperation, ValueError) as exc:
+        raise RgbdConversionError(
+            f"{source} line {line_number}: invalid timestamp {text!r}"
+        ) from exc
```

Ruff (0.16.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)